WikiWatershed/model-my-watershed
# Switch Climate Analysis to Multioperation

## Overview

To reduce Celery overhead and the possibility of unchecked `chord_unlock` calls, we switch Climate analysis to the multioperation geoprocessing call. This results in a more efficient, faster analysis that is easier to cache.

Connects #2782 

### Demo

![image](https://user-images.githubusercontent.com/1430060/71631563-48f1f180-2bd7-11ea-911c-d80252f3432a.png)

## Testing Instructions

* Check out this branch and debug celery `$ ./scripts/debugcelery.sh`
* Select a WKAOI (preferably a fresh one that would not have been cached on your local machine) and analyze it
    - [x] Ensure all analyses complete successfully, including Climate
* Select the same WKAOI on staging
    - [x] Ensure the Climate analysis has the same numbers as your local
* Also try with drawing a shape, so it doesn't use the cache
    - [x] Ensure it completes successfully
